packages/frontend/web/src/components/form-cocktail/ToppingPart.tsx

```diff
@@ -1,17 +1,109 @@
-import type { ToppingPartProps } from '@app/types';
+import { MoveRight, Skull } from 'lucide-react';
+import { useEffect, useMemo, useState } from 'react';
 
-const toppings = ['shrimps', 'lemon', 'mint'];
+import type { Topping, ToppingPartProps } from '@app/types';
 
 export default function ToppingPart({
   register,
   selectedTopping,
+  selectedIngredient,
+  selectedAlcohol,
   handleToppingChange,
   errors,
 }: ToppingPartProps) {
+  const [toppings, setToppings] = useState<Topping[]>([]);
+  const [mainFlavour, setMainFlavour] = useState<string>('');
+  const [randomTopping, setRandomTopping] = useState<Topping | null>(null);
+  const [isRandomToppingChoosen, setIsRandomToppingChoosen] =
+    useState<boolean>(false);
+  const [isShowRandomTopping, setIsShowRandomTopping] =
+    useState<boolean>(false);
+
+  const memorizedFlavoursCount = useMemo(() => {
+    const allFlavours = [selectedAlcohol?.flavour, selectedIngredient?.flavour];
+
+    // eslint-disable-next-line unicorn/no-array-reduce
+    return allFlavours.reduce<Record<string, number>>(
```

Comment:
Tu peux pas mettre 
```suggestion
    return allFlavours.reduce<Record<Flavour, number>>(
```
ici ? Tu pourrais avoir de l'autocomplétion grâce à ça